### ISSUE TITLE
in screen1, setState not triggered after moving to screen 2 and  return to it

### ISSUE THREAD

**Issue by ibrahimsenan** [aksonov/react-native-router-flux#2952]
### Version
Tell us which versions you are using: 

I am using 
 "react-native-router-flux": "^4.0.0-beta.27",
 "react": "^16.0.0-alpha.12",
 "react-native": "^0.46.1",

### Expected behaviour
Component re-render after setState.

### Actual behaviour
When I go from one screen to another, the first screen stops responding to any setState command. So any changes in the UI after setState is not working while it was working in the first time before moving to another screen.

**Comment by Blapi** [comment]
It is not related to this library, you must do something wrong in your code. If you still have this problem, go on StackOverflow and check examples on internet. setState works fine as you can see on the example project

**Comment by ibrahimsenan** [comment]
@Blapi I have checked the code many times  but the screen never update or change the view after I navigate to another one and get back to it, it got disabled some home, I am depending on socket to update the status, do you have any idea about that.

@Blapi thank you

**Comment by Blapi** [comment]
What exactly are you doing when you're rendering a new Scene that is not is the expacted behaviour ?

Show some code (formatted) please, and detail your navigation flow. It is very unclear.

**Comment by Blapi** [comment]
If this is about `Scene 1 -> Scene 2 -> Scene 1` and that, when going back to your Scene 1, there should be a change, you should check https://reactjs.org/docs/react-component.html#componentwillreceiveprops, it should help you in your case.

Of course, it depends on how you handle your store and props.

**Comment by Blapi** [comment]
Also, as you can see in the doc :


onEnter | Function |   | Called when the Scene is navigated to. props are provided as a function param. Only scenes with 'component' defined is supported. Your component class may also have onEnter function
-- | -- | -- | --

**Comment by ibrahimsenan** [comment]
The problem on first initializing the project I have Scene 1 with drawer, when I navigate from drawer to  Scene  2 and then back to --> Scene 1, if I get any socket or SetState it will not update in Scene 1 

![video_2018-03-22_174428-00_00_00-00_00_26](https://user-images.githubusercontent.com/15571298/37771182-d03200de-2dfc-11e8-9222-c00e9e7f3d84.gif)

**Comment by Blapi** [comment]
I guess you don't handle your store properly, are you using redux or some equivalent ?

If no, you should consider using it, you will easily fix your problem as you could use your orders as props and map these props to your state, re-rendering your Scene 1.

If yes, you're doing something wrong.

**Comment by ibrahimsenan** [comment]
I did not understand, why Scene  1, interact well at the first time and it updates the UI and it showing the new orders as received from the socket normally, but when I go to other Sence from the drawer,  Scene 1 stop updating the UI totally, even when the state changes but Ui not responding, Listview it stopped totally. I tried to do ForceUpdate but nothing worked. 

Thank you @Blapi  for your great help.
